pkg/core/resources/model/resource.go

```diff
@@ -71,6 +73,12 @@ type ResourceValidator interface {
 }
 
 func Validate(resource Resource) error {
+	if resource.GetSpec() != nil {
+		if err := plugins.ValidateResourceSchema(resource.GetSpec(), string(resource.Descriptor().Name)); err != nil {
```

Comment:
Why do we want to call schema validation here? I think at this point validation already passed in `rest.Unmarshal`, isn't it?

Reply:
see https://github.com/kumahq/kuma/pull/5107#discussion_r991141032